src/v/model/record_batch_reader.cc

```diff
@@ -135,6 +135,22 @@ record_batch_reader make_foreign_memory_record_batch_reader(record_batch b) {
     return make_foreign_memory_record_batch_reader(std::move(data));
 }
 
+record_batch_reader make_empty_record_batch_reader() {
+    class reader final : public record_batch_reader::impl {
+    public:
+        bool is_end_of_stream() const final { return true; }
+
+        ss::future<storage_t> do_load_slice(timeout_clock::time_point) final {
+            co_return data_t{};
+        }
+
+        void print(std::ostream& os) final {
+            fmt::print(os, "{empty_record_batch_reader}");
```

Comment:
won't fmtlib treat `{empty_record_batch_reader}` as a named argument? I don't think these are included in compile time checks, so it might only fail if you call it...

Reply:
Hm yea seems like this should just be `os << "{empty_record_batch_reader}";`

Reply:
thanks. fixed

Reply:
Or `fmt::print(os, "{{empty_record_batch_reader}}")` or something. Or maybe it works as written.